python/lsst/daf/butler/registry/queries/_builder.py

```diff
@@ -224,7 +216,7 @@ def joinDataset(
                             f"collection '{collectionRecord.name}' is not yet supported."
                         )
                     else:
-                        calibration_collections.append(collectionRecord)
+                        collectionRecords.append(collectionRecord)
                 else:
                     # We can never find a non-calibration dataset in a
                     # CALIBRATION collection.
```

Comment:
Re-reading pre-existing code - I think this comment is not quite right, in this `else` branch dataset type can still be calibration (if `collectionRecord.name` is not in `explicitCollections`).

Reply:
Yes, I believe this was an edge-case bug that is now fixed, and the comment is now correct.